http/pom.xml

```diff
@@ -13,7 +13,7 @@
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <java.version>17</java.version>
         <azure.functions.maven.plugin.version>1.36.0</azure.functions.maven.plugin.version>
-        <azure.functions.java.library.version>2.2.0</azure.functions.java.library.version>
+        <azure.functions.java.library.version>3.1.0</azure.functions.java.library.version>
```

Comment:
core tools is generating 1.24 functions java library pom.xml. core tool templates need an update. is this tested with 3.1.0 version?

Reply:
This sample itself is.  I think we need to do the right thing despite what the templates are doing.  We should not be shipping 2.2.0 in templates!  Those are from 2022.  3.1 has been out there for 9 months.  Let's use it.
https://central.sonatype.com/artifact/com.microsoft.azure.functions/azure-functions-java-library/versions

Reply:
Yes, I agree maven repo has 3.1 as well - [Maven Repository: com.microsoft.azure.functions » azure-functions-java-library (mvnrepository.com)](https://mvnrepository.com/artifact/com.microsoft.azure.functions/azure-functions-java-library)

LGTM !